libs/knowledge-store/ragstack_knowledge_store/knowledge_store.py

```diff
@@ -1,49 +1,67 @@
 import secrets
-from dataclasses import dataclass
+from dataclasses import dataclass, field
+from enum import Enum
 from typing import (
-    Any,
     Dict,
     Iterable,
     List,
     NamedTuple,
     Optional,
     Sequence,
     Tuple,
-    Type,
 )
 
 import numpy as np
 from cassandra.cluster import ConsistencyLevel, ResponseFuture, Session
 from cassio.config import check_resolve_keyspace, check_resolve_session
-from langchain_community.utilities.cassandra import SetupMode
-from langchain_community.utils.math import cosine_similarity
-from langchain_core.documents import Document
-from langchain_core.embeddings import Embeddings
-
-from ragstack_knowledge_store.edge_extractor import get_link_tags
 
 from ._utils import strict_zip
-from .base import KnowledgeStore, Node, TextNode
 from .concurrency import ConcurrentQueries
 from .content import Kind
+from .edge_extractor import get_link_tags
+from .embedding_model import EmbeddingModel
+from .math import cosine_similarity
 
 CONTENT_ID = "content_id"
 
 
-def _row_to_document(row) -> Document:
-    return Document(
-        page_content=row.text_content,
+@dataclass
+class Node:
+    """Node in the KnowledgeStore graph"""
+
+    id: Optional[str] = None
+    """Unique ID for the node. Shall be generated by the KnowledgeStore if not set"""
+    metadata: dict = field(default_factory=dict)
+    """Metadata for the node. May contain information used to link this node 
+    with other nodes."""
+
+
+@dataclass
+class TextNode(Node):
+    text: str = None
+    """Text contained by the node"""
+
+
+class SetupMode(Enum):
```

Comment:
I wonder if this is too much LangChain-ism in the base. What if we just had methods for setup (`setup_schema(...)`, `asetup_schema(...)`, etc.). Then the langchain code can choose which to call based on the setup mode from LangChain, but we don't need to mirror the enum into base?

Reply:
We could. But our strategy until now in cassio/astrapy/etc... has been to default to creating the schema in the constructor to minimize the number of operations to do in notebooks.
We can change that here.